src/MLJInterface.jl

```diff
@@ -389,17 +393,284 @@ MLJModelInterface.metadata_model(
     path="LightGBM.MLJInterface.LGBMClassifier",
     input=MLJModelInterface.Table(MLJModelInterface.Continuous),
     target=AbstractVector{<:MLJModelInterface.Finite},
-    weights=true,
-    descr="Microsoft LightGBM FFI wrapper: Classifier",
+    weights=true
+    # descr="Microsoft LightGBM FFI wrapper: Classifier",
 )
 
 MLJModelInterface.metadata_model(
     LGBMRegressor; # end positional args
     path="LightGBM.MLJInterface.LGBMRegressor",
     input=MLJModelInterface.Table(MLJModelInterface.Continuous),
     target=AbstractVector{MLJModelInterface.Continuous},
-    weights=true,
-    descr="Microsoft LightGBM FFI wrapper: Regressor",
+    weights=true
+    # descr="Microsoft LightGBM FFI wrapper: Regressor",
 )
 
+"""
+`LightGBMRegressor`: LightGBM, short for light gradient-boosting machine, is a framework for gradient boosting
+based on decision tree algorithms and used for ranking, classification and other machine
+learning tasks, with a focus on performance and scalability. This model in particular is used for various types
+of regression
+
+# Training data
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
+Here:
+- `X` is any table of input features (eg, a `DataFrame`) whose columns are of scitype
+  `Continuous`; check the column scitypes with `schema(X)`; alternatively, `X` is any
+  `AbstractMatrix` with `Continuous` elements; check the scitype with `scitype(X)`.
+- y is a vector of targets whose items are of scitype `Continuous`. Check the scitype with
+  scitype(y).
+
+Train the machine using `fit!(mach, rows=...)`.
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+  features `Xnew` having the same Scitype as `X` above.
+
+# Hyper-parameters
+
+- `boosting::String = "gbdt"`: Which boosting algorithm to use. One of:
```

Comment:
Did you write all of these up yourself or did you obtain them from somewhere? I ask because I'm interested in the maintenance effort

Reply:
I wrote them all up using the [LightGBM docs](https://lightgbm.readthedocs.io/en/latest/Parameters.html) as a reference!

Reply:
Thanks for clarifying

Reply:
Also, not to trivialise the effort that would have taken, thanks for the effort!